### PR TITLE
fix: change target's type in fee estimate map

### LINKED ISSUE #64: Keys in feerates mapping should be `u16`, not `String`.
In `get_fee_estimates` the confirmation targets are passed as strings:
https://github.com/bitcoindevkit/rust-esplora-client/blob/ef1925e1ee1b951f15a9339282858bea27c6168f/src/blocking.rs#L344-L365

But they are given as `u16` by `electrs`: https://github.com/mempool/electrs/blob/b379d24cb1972976d4b787d6e4c0f664f144bfa6/src/new_index/query.rs#L196-L205 (mempool's fork), https://github.com/Blockstream/electrs/blob/fd35014283c7d3a7a85c77b9fd647c9f09de12c9/src/new_index/query.rs#L184-L193 (Blocktream's fork).

### PR DESCRIPTION
**Pull request by vladimirfomene**
Fixes #64.

## Summary by CodeRabbit

- **Refactor**
	- Updated the method for fetching fee estimates to use more efficient data types.
	- Improved the fee rate conversion process for enhanced accuracy and performance.

### CODE REVIEW

**Comment by oleonardolima** [comment]
@vladimirfomene I got curious, did the CI suddenly start failing?

**Comment by oleonardolima** [comment]
> @oleonardolima is failing because some dependencies have been updated so we have to pin them to make sure we are able to build with our MSRV

Oh, thanks! I saw that the rustls dependency version got yanked, and lead to this 😅

**Review by oleonardolima — APPROVED** [review]
utACK

**Comment by oleonardolima** [comment]
@vladimirfomene through a re-review and some manual tests here, I saw that an fn and a test still expect the old parameter signature, and would need to be updated as well:
- `convert_fee_rate` fn: https://github.com/bitcoindevkit/rust-esplora-client/blob/ef1925e1ee1b951f15a9339282858bea27c6168f/src/lib.rs#L81-L97
- `feerate_parsing` test: https://github.com/bitcoindevkit/rust-esplora-client/blob/ef1925e1ee1b951f15a9339282858bea27c6168f/src/lib.rs#L333-L375

File: .github/workflows/cont_integration.yml [hunk 1]
```diff
@@ -55,16 +55,22 @@ jobs:
...
+        cargo update -p rustls-webpki:0.101.7 --precise 0.101.1
         cargo update -p tempfile --precise 3.6.0
         cargo update -p h2 --precise 0.3.20
-        cargo update -p flate2:1.0.27 --precise 1.0.26
+        cargo update -p flate2:1.0.28 --precise 1.0.26
+        cargo update -p webpki:0.22.4 --precise 0.22.2
+        cargo update -p sct --precise 0.7.0
+        cargo update -p byteorder --precise 1.4.3
+        cargo update -p log --precise 0.4.18
+        cargo update -p jobserver --precise 0.1.26
```
Comment (LLFourn) [inline .github/workflows/cont_integration.yml:69]: Can all this kind of shit be removed now that we bumped MSRV?

**Comment by notmandatory** [comment]
Please rebase to pick up changes in #69 that fix CI.

**Review by oleonardolima — COMMENTED** [review]
utACK

**Review by oleonardolima — APPROVED** [review]
ACK 3add7f9a8af5bfa70c27140788f3c1a6ced9b0f7

**Review by notmandatory — APPROVED** [review]
utACK 3add7f9a8af5bfa70c27140788f3c1a6ced9b0f7

**Comment by notmandatory** [comment]
FYI I had to rebase this since #75 was merged.

**Pull request merged**